From 6bf806918027147a1d54c23109582b976295a40d Mon Sep 17 00:00:00 2001
From: Fei Wang <fei.w.wang@intel.com>
Date: Tue, 21 Jun 2022 13:49:48 +0800
Subject: [PATCH 37/46] fftools/ffmpeg: don't assert when EAGAIN in flushing
 encoder

When flush encoder, there may have cached frames/packets in encoder
which can still be received, and the encoder doesn't need to guarantee
an available output in each receive_packet in additional.

Signed-off-by: Fei Wang <fei.w.wang@intel.com>
---
 fftools/ffmpeg.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fftools/ffmpeg.c b/fftools/ffmpeg.c
index e57486fd4a..c59e01cff0 100644
--- a/fftools/ffmpeg.c
+++ b/fftools/ffmpeg.c
@@ -864,7 +864,8 @@ static int encode_frame(OutputFile *of, OutputStream *ost, AVFrame *frame)
             fprintf(ost->logfile, "%s", enc->stats_out);
 
         if (ret == AVERROR(EAGAIN)) {
-            av_assert0(frame); // should never happen during flushing
+            if (!frame)
+                continue;
             return 0;
         } else if (ret == AVERROR_EOF) {
             of_output_packet(of, pkt, ost, 1);
-- 
2.34.1

